tensorboard/plugins/profile/pod_viewer/topology_graph/topology-graph.ts

```diff
@@ -0,0 +1,594 @@
+   * Draw the UI of node cards.
+   */
+  drawNodeCards: function(svg: any, data: Array<TopoData>, colorScale: any) {
+    let parent = this;
+    let cards = svg.selectAll('.xdim').data(data, (d) => d.xdim);
+    cards.enter()
+         .append('rect')
+         .attr('id', (d) => 'rid' + d.rid)
+         .attr('x', (d) => parent.getNodeXLoc(
+                             Math.floor(d.xdim / parent._hostXStride),
+                               d.xdim % parent._hostXStride, d.nid))
+         .attr('y', (d) => parent.getChipYLoc(
+                             Math.floor(d.ydim / HOST_Y_STRIDE),
+                               d.ydim % HOST_Y_STRIDE))
+         .attr('rx', 4 / NODES_PER_CHIP)
+         .attr('ry', 4)
+         .attr('class', 'hour bordered')
+         .attr('width', parent._nodeGridWidth)
+         .attr('height', parent._nodeGridHeight)
+         .attr('border', 1)
+         .style('stroke', 'black')
+         .style('stroke-width', 1)
+         .style('fill', (d) => colorScale(d.value / d.total))
+         .merge(cards)
+         .on('mouseover',
+             function(d) {
+               // highlight text
+               d3.select(this)
+                 .classed('cell-hover', true)
+                 .style('opacity', 0.5);
+
+               // Update the tooltip position and value
+               d3.select(parent.$.tooltip)
+                 .style('left', d3.event.pageX + 10 + 'px')
+                 .style('top', d3.event.pageY - 10 + 'px')
+                 .select('#value')
+                 .text(parent._getToolTipText(d));
+               d3.select(parent.$.tooltip)
+                 .classed('hidden', false);
+             })
+         .on('mouseout',
+             function() {
+               d3.select(this)
+                 .classed('cell-hover', false)
+                 .style('opacity', 1.0);
+              d3.select(parent.$.tooltip).classed('hidden', true);
+             });
+    cards.exit().remove();
+  },
+  /**
+   * Draw the UI of chip to chip links.
+   */
+   drawLinks: function(linkData: Array<podviewer.proto.ChannelInfo>) {
+    let parent = this;
+    if (!linkData || linkData.length == 0 || !this._gLink) {
+      return;
+    }
+
+    // Handle links;
+    let links = this._gLink.selectAll('.link').data(linkData);
+
+    // attach the arrow from defs
+    links.enter()
+         .append('svg:path')
+         .attr('id', (d) => 'cid' + d.channelId)
+         .attr('stroke-width', 2)
+         .attr('stroke', 'red')
+         .attr('fill', 'none')
+         .attr('marker-end', 'url(#arrow)')
+         .style('visibility', 'hidden')
+         .merge(links)
+         .attr('d', (d) => parent.linkToPath(d));
+
+    // Handle deleted links.
+    links.exit().remove();
+    this._selectedChannelIdChanged(this.selectedChannelId);
+  },
+  /**
+   * Given the global core id, returns the (x, y) coordinates in the topology
+   * graph.
+   */
+  coreIdToPos: function(id: number): Position {
+    let p = this;
```

Comment:
this short notation is discouraged (have more meaningful variable name) and unnecessary. Consider removing it.

Reply:
Done.